### DISCUSSION TITLE
Column level lineage

### DISCUSSION THREAD

**Discussion by toddy86** [dagster-io/dagster#19884] [Q&A] [answered]
Hi, 

Are there any plans for column level data lineage in Dagster? 

The asset level lineage is already great. And what would make it even better is column level lineage. 

As an example of where this would be useful for my current use case, each user in our database is attributed with a tag. The initial tag is set when the user is created. However, we re-attribute users to the correct tag later in the day once a reattribution job has run.

So at present, using asset level lineage we rematerialise all of the downstream assets that use the source attribution dataset. But this results in a greedy materialisation policy because it's only a single column which is updated and not required in all of the downstream assets.

Thanks

Todd

**Comment by rexledesma** [comment, marked as answer]
We have documentation for materializing assets with column-level lineage here: https://docs.dagster.io/concepts/metadata-tags/asset-metadata/column-level-lineage.

You can view this column-level lineage in Dagster+.

We don't currently have a feature that enhances our materialization logic using this column-level lineage metadata. Feel free to file a feature request for that.
